js/ui/grid_core/ui.grid_core.validating.js

```diff
@@ -1289,10 +1290,13 @@ export default {
                         rowKey: oldRow.key,
                         columnIndex
                     });
+                    const validationData = validatingController._getValidationData(oldRow.key);
                     const newValidationStatus = validationResultIsValid(validationResult) ? validationResult.status : validationResult;
                     const rowIsModified = JSON.stringify(newRow.modifiedValues) !== JSON.stringify(oldRow.modifiedValues);
                     if(oldValidationStatus !== newValidationStatus && rowIsModified) {
                         return true;
+                    } else if(validationData.isValid) {
+                        $(cell?.cellElement).toggleClass(this.addWidgetPrefix(INVALIDATE_CLASS), false);
```

Comment:
I think that the isCellChanged method should not change anything. 
```suggestion
                    const cellIsMarkedAsInvalid = $(cell?.cellElement).hasClass(this.addWidgetPrefix(INVALIDATE_CLASS));
                    if((oldValidationStatus !== newValidationStatus && rowIsModified) || (validationData.isValid && cellIsMarkedAsInvalid)) {
                        return true;
                    }
```